### PR TITLE
Pal 259 logging improvements for audit service

### PR DESCRIPTION
**Pull request by developer01189998819991197253**
As this may be the first of many similar, it is worth ensuring the initial set of changes to the user service are complete

The ticket suggests:

Log at INFO level for receipt of a message
Log at INFO level for a message sent to another service
Log at DEBUG for individual function calls with more metadata
In the event of ERRORs include the METAdata of what was being handled
If this is the full set of cases for logging, we should only see messages on INFO if services are communicating with one another

### CODE REVIEW

**Review by dev-958 — CHANGES_REQUESTED** [review]
It's overkill to log (even in debug) setters/getters in data objects. Yes log their contents once populated but it will render the logs unreadable.

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/request/AuditRequest.java [hunk 1]
```diff
@@ -51,28 +53,33 @@
...
 public class AuditRequest<ZoneDateTime> extends Request {
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(AuditRequest.class);
+
     public final ZonedDateTime timestamp;
     public final String serverIp;
     public final String serverHostname;
 
     protected AuditRequest() {
+        LOGGER.debug("AuditRequest called");
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/request/AuditRequest.java]: Logging every constructor call is overkill

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java [hunk 2]
```diff
@@ -286,15 +312,18 @@ private static void onReadRequestException(final DefaultEventLoggingService logg
      */
     public StroomAuditService systemName(final String systemName) {
         requireNonNull(systemName, "The system name cannot be null.");
+        LOGGER.debug("systemName is {}", systemName);
         SYSTEM.setName(systemName);
         return this;
     }
 
     public String getSystemName() {
+        LOGGER.debug("systemName is {}", SYSTEM.getName());
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java]: We don't need to log every getter and setter.

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java [hunk 3]
```diff
@@ -286,15 +312,18 @@ private static void onReadRequestException(final DefaultEventLoggingService logg
...
         return SYSTEM.getName();
     }
 
     public void setSystemName(final String systemName) {
+        LOGGER.debug("systemName is {}", systemName);
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java]: This method is simply calling the other one which is already logging.

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java [hunk 4]
```diff
@@ -322,15 +354,18 @@ public void setOrganisation(final String organisation) {
...
         return this;
     }
 
     public String getSystemEnv() {
+        LOGGER.debug("systemEnv is {}", SYSTEM.getEnvironment());
         return SYSTEM.getEnvironment();
     }
 
     public void setSystemEnv(final String systemEnv) {
+        LOGGER.debug("systemEnv is {}", systemEnv);
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java]: As above

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java [hunk 5]
```diff
@@ -322,15 +354,18 @@ public void setOrganisation(final String organisation) {
      */
     public StroomAuditService systemEnv(final String env) {
         requireNonNull(env, "The env cannot be null.");
+        LOGGER.debug("systemEnv is {}", env);
         SYSTEM.setEnvironment(env);
         return this;
     }
 
     public String getSystemEnv() {
+        LOGGER.debug("systemEnv is {}", SYSTEM.getEnvironment());
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java]: Don't log every getter

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java [hunk 6]
```diff
@@ -340,15 +375,18 @@ public void setSystemEnv(final String systemEnv) {
...
         return this;
     }
 
     public String getSystemDescription() {
+        LOGGER.debug("systemDescription is {}", SYSTEM.getDescription());
         return SYSTEM.getDescription();
     }
 
     public void setSystemDescription(final String description) {
+        LOGGER.debug("systemDescription is {}", description);
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java]: ditto

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java [hunk 7]
```diff
@@ -358,15 +396,18 @@ public void setSystemDescription(final String description) {
...
         return this;
     }
 
     public String getSystemVersion() {
+        LOGGER.debug("systemVersion is {}", SYSTEM.getVersion());
         return SYSTEM.getVersion();
     }
 
     public void setSystemVersion(final String systemVersion) {
+        LOGGER.debug("systemVersion is {}", systemVersion);
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java]: Ditto

File: audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java [hunk 8]
```diff
@@ -376,17 +417,20 @@ public void setSystemVersion(final String systemVersion) {
...
         return this;
     }
 
     public String getSystemClassification() {
+        LOGGER.debug("systemClassification is {}", SYSTEM.getClassification().getText());
         return SYSTEM.getClassification().getText();
     }
 
     public void setSystemClassification(final String systemClassification) {
+        LOGGER.debug("systemClassification is {}", systemClassification);
```
Comment (m09526) [inline audit-service/src/main/java/uk/gov/gchq/palisade/service/audit/service/StroomAuditService.java:424]: As above

**Review by m09526 — CHANGES_REQUESTED** [review]

**Review by m09526 — APPROVED** [review]

**Review by dev-958 — APPROVED** [review]

**Pull request merged**